### PR TITLE
Avishay/mgmt 1099

### PR DESCRIPTION
**Pull request by RazRegev**
rebased from https://github.com/filanov/bm-inventory/pull/254

### CODE REVIEW

**Comment by eranco74** [comment]
is this relevant?
I see the same review here: https://github.com/filanov/bm-inventory/pull/254

File: docs/ClusterState.dot [hunk 1]
```diff
@@ -12,11 +12,19 @@ digraph ClusterStateMachine {
...
-    ready -> insufficient [style=invis];
+    error -> insufficient [label = "reset", color=lightpink3, fontcolor=lightpink3];
     ready -> installing [label = "installation started", color=darkolivegreen4, fontcolor=darkolivegreen4];
     installing -> installed [label = "installation\ncompleted", color=chartreuse4, fontcolor=chartreuse4];
+<<<<<<< HEAD
+    installing -> error [label = "installation\nerror\nor canceled", color=crimson, fontcolor=crimson];
+=======
     installing -> error [label = "installation\nerror", color=crimson, fontcolor=crimson];
+    error -> insufficient [label = "reset\nstate", color=lightpink3, fontcolor=lightpink3];
+>>>>>>> 014a74c... MGMT-1099: Add api to reset cluster state (swagger+docs)
```
Comment (ronniel1) [inline docs/ClusterState.dot:24]: Need to fix this

File: docs/ClusterState.dot [hunk 2]
```diff
@@ -12,11 +12,19 @@ digraph ClusterStateMachine {
...
```
Comment (avishayt) [inline docs/ClusterState.dot:23]:
```suggestion
    installing -> error [label = "installation\nerror\nor canceled", color=crimson, fontcolor=crimson];
```

File: docs/NodeState.dot [hunk 3]
```diff
@@ -15,30 +15,32 @@ digraph NodeStateMachine {
...
 
     disabled -> discovering [label = "user\nenables", color=goldenrod, fontcolor=goldenrod];
 
-    //insufficient -> disabled [label = "user\ndisables", color=grey29, fontcolor=grey29];
     insufficient -> known [label = "sufficient\nfor role", color=cadetblue4, fontcolor=cadetblue4];
 
     installing -> installed [label = "installation\ncompleted", color=chartreuse4, fontcolor=chartreuse4];
     installing -> error [label = "installation\nerror", color=crimson, fontcolor=crimson];
 
+    error -> resetting [label = "reset", color=aquamarine4, fontcolor=aquamarine4];
```
Comment (filanov) [inline docs/NodeState.dot:40]: How host monitor will be affected by this state?

File: internal/host/resetting.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+	return nil, errors.Errorf("unable to update hardware info to host <%s> in <%s> status",
+		h.ID, swag.StringValue(h.Status))
+}
+
+func (r *resettingState) UpdateInventory(ctx context.Context, h *models.Host, inventory string) (*UpdateReply, error) {
+	return nil, errors.Errorf("unable to update inventory to host <%s> in <%s> status",
+		h.ID, swag.StringValue(h.Status))
+}
+
+func (r *resettingState) UpdateRole(ctx context.Context, h *models.Host, role string, db *gorm.DB) (*UpdateReply, error) {
```
Comment (filanov) [inline internal/host/resetting.go:32]: you don't need this function anymore.

**Comment by RazRegev** [comment]
> Not sure that i understand the flow. how do we get out or `resetting` host state?

After this PR will be merged, this PR https://github.com/filanov/bm-inventory/pull/308 meant to get us out of the resetting state by telling the agents to reset themselves and re-register.

**Pull request merged**